--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,8 @@
 
 # Options
 option(RUN_NINJA_TEST "Run Ninja test suite" OFF)
+option(USE_PREBUILT_NINJA_BINARY "Whether to use the prebuilt ninja executable file provided by user" OFF)
+set(PREBUILT_NINJA_PATH "/usr/bin/ninja" CACHE PATH "The path of the prebuilt ninja binary")
 
 message(STATUS "************************************")
 message(STATUS "Ninja Python Distribution")
@@ -23,6 +25,7 @@
 message(STATUS "  RUN_NINJA_TEST : ${RUN_NINJA_TEST}")
 message(STATUS "************************************")
 
+if(NOT USE_PREBUILT_NINJA_BINARY)
 add_subdirectory(ninja-upstream)
 
 if(RUN_NINJA_TEST)
@@ -35,3 +38,10 @@
 
 install(TARGETS ninja COMPONENT python DESTINATION "${SKBUILD_SCRIPTS_DIR}")
 install(FILES "${ninja_SOURCE_DIR}/misc/ninja_syntax.py" COMPONENT python DESTINATION ninja)
+else()
+if(NOT EXISTS ${PREBUILT_NINJA_PATH})
+  message(FATAL_ERROR "PREBUILT_NINJA_PATH doesn't exist!")
+endif()
+install(PROGRAMS ${PREBUILT_NINJA_PATH} COMPONENT python DESTINATION "${SKBUILD_SCRIPTS_DIR}")
+install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/ninja-upstream/misc/ninja_syntax.py" COMPONENT python DESTINATION ninja)
+endif()
