--- a/setup.py
+++ b/setup.py
@@ -78,4 +78,10 @@
     keywords='ninja build c++ fortran cross-platform cross-compilation',
 
     extras_require={"test": test_requirements},
+
+    use_scm_version={
+        "version_scheme": lambda version:version.format_with("{tag}"),
+        "local_scheme": "no-local-version",
+        "write_to": "src/ninja/_version.py",
+    },
 )
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -6,9 +6,6 @@
 ]
 build-backend = "setuptools.build_meta"
 
-[tool.setuptools_scm]
-write_to = "src/ninja/_version.py"
-
 [tool.cibuildwheel]
 build = "cp39-*"
 build-verbosity = 1
